packages/worker/src/workflow/bundler.ts

```diff
@@ -105,7 +105,10 @@ export class WorkflowCodeBundler {
   protected async bundle(filesystem: typeof unionfs.ufs, entry: string, distDir: string): Promise<void> {
     const compiler = webpack({
       resolve: {
-        modules: this.nodeModulesPaths,
+        modules: [
+          ...this.nodeModulesPaths,
+          ...(!this.nodeModulesPaths.includes('node_modules') ? ['node_modules'] : []),
```

Comment:
Is this `node_modules` relative to PWD?
What happens if you run this from a different directory?
An ideal solution would take that into account.

Reply:
It is _not_ relative to PWD. You may want to look at [Webpack's doc](https://webpack.js.org/configuration/resolve/#resolvemodules) for an explanation on this.

To be honest, I feel they are mixing two different concepts here. But essentially, adding the string `node_modules` in this array is required for Webpack to resolve modules the same way as Node does it (that is, if file /a/b/c/d.js request module x, then look at /a/b/c/node_modules/x, then /a/b/node_modules/x, , then /a/node_modules/x).

Reply:
The docs don't specify what it's relative to that's why I assumed PWD. What else would it be? The entry point?

Reply:
To put it simply, it is relative to every source file that request a module... and to all parent directories of that source file. Well, not exactly simple when I phrase it like this, eh? 🤣 

What that means is that assuming the following configuration:

```
resolve: {
    modules: ['/Users/myself/myproject/MagicDirectory', 'node_modules']
}
```

and assuming that webpack is processing file A, located at `/Users/myself/myproject/src/submodule/a.js`, which imports module `B`, webpack will look for the following locations, in that specific order:

```
/Users/myself/myproject/MagicDirectory/B/package.json
/Users/myself/myproject/src/submodule/node_modules/B/package.json
/Users/myself/myproject/src/node_modules/B/package.json
/Users/myself/myproject/node_modules/B/package.json
/Users/myself/node_modules/B/package.json
/Users/node_modules/B/package.json
/node_modules/B/package.json
```

The search succeeds as soon as it finds an entry that exists.

Now, let's suppose that B was indeed found, at `/Users/myself/myproject/src/submodule/node_modules/B/package.json`. And suppose that the package.json file specify that the entrypoint of that module is at `lib/index.js`, and requires module `C`.

Now, webpack tries to resolve module `C` from the location of module B's index.js. That means that webpack will look for the following locations:

```
/Users/myself/myproject/MagicDirectory/C/package.json
/Users/myself/myproject/src/submodule/node_modules/B/lib/node_modules/C/package.json
/Users/myself/myproject/src/submodule/node_modules/B/node_modules/C/package.json
/Users/myself/myproject/src/submodule/node_modules/node_modules/C/package.json
/Users/myself/myproject/src/submodule/node_modules/C/package.json
/Users/myself/myproject/src/node_modules/C/package.json
/Users/myself/myproject/node_modules/C/package.json
/Users/myself/node_modules/C/package.json
/Users/node_modules/C/package.json
/node_modules/C/package.json
```

Obviously, most of these attempts are ridiculous, but a few are critical, notably:

`/Users/myself/myproject/MagicDirectory/C/package.json`: We explicitly specified the `MagicDirectory`, so there might be some important things there...
`/Users/myself/myproject/src/submodule/node_modules/B/node_modules/C/package.json`: This is where NPM will place package C (assuming it is a requirement of B) if that package has not been hoisted.
`/Users/myself/myproject/node_modules/C/package.json`: This is where NPM will place package C if it has been hoisted (NPM tends to hoist whenever it is possible)

Reply:
Got it, this makes sense now and could actually explain the issues I was seeing with pnpm and hoisting (#421).
If this is the behavior then the fix is pretty good, and I'd go as far as removing the [default `nodeModulesPaths` we inject in the worker](https://github.com/temporalio/sdk-typescript/blob/7ad8b1a74b5217613d2bfdd7a716b7e2e61bf1bc/packages/worker/src/worker-options.ts#L316).

Seems like putting the entrypoint next to `workflowsPath` removes the need to specify `resolve.modules` altogether so I'd say it's a preferred approach.